NBi.Services/TestCaseManager.cs

```diff
@@ -54,7 +54,16 @@ public void ReadFromQuery(string sql, string connectionString)
                 variables.Add(col.ColumnName);
         }
 
-        private DataTable content;
+        public void ReadFromDataTable(DataTable datatable)

```

Comment:
It's true that I wrote other ReadFrom methods in the ```State``` class but they should be in the ```Action``` class. No problem, I need to refactor this and will do it.